src/frontend/picker.ts

```diff
@@ -0,0 +1,158 @@
+        this.name = name + "["+id+"]";
+        this.id = id;
+        this.image = image
+        this.labels = labels;
+    }
+    
+    toAttachItem() {
+        return {
+            label: this.name,
+            description: this.labels,
+            detail: this.image,
+            id: this.id
+        };
+    };
+};
+
+class ProcessItem implements IItem{
+    name: string;
+    id: string;
+    protected commandLine: string;
+
+    constructor(name, pid, commandLine) {
+        this.name = name;
+        this.id = pid;
+        this.commandLine = commandLine;
+    }
+
+    toAttachItem() {
+        return {
+            label: this.name,
+            description: this.id,
+            detail: this.commandLine,
+            id: this.id
+        };
+    };
+};
+
+abstract class Picker{
+    launchConfig: any;
+
+    getItems(){
+        return this.getEntries().then(function (processEntries: any[]) {
+            return processEntries.map(function (p) { return p.toAttachItem(); });
+        });
+    };
+
+    pickItem(launchConfig){
+        this.launchConfig = launchConfig;
+
+        var attachPickOptions = {
+            matchOnDescription: true,
+            matchOnDetail: true,
+            placeHolder: "Select the docker name to attach to"
+        };
+        return vscode.window.showQuickPick(this.getItems(), attachPickOptions)
+            .then(function (dockerItem: any) {
+                return dockerItem ? dockerItem.id : null;
+            });
+    };
+
+    parseItems(items: string){
+    //Parses a line of the output and returns array of objects
+        var lines = items.split('\n');
+        var entries = [];
+        for (var i = 1; i < lines.length; i++) {
+            var line = lines[i];
+            if (!line) {
+                continue;
+            }
+            var docker_1 = this.lineParser(line);
+            entries.push(docker_1);
+        }
+        return entries;
+    }
+
+    execChildProcess(process, workingDirectory) {
+        return new Promise(function (resolve, reject) {
+            child_process.exec(process, { cwd: workingDirectory, maxBuffer: 512000 }, function (error, stdout, stderr) {
+                if (error) {
+                    reject(error);
+                    return;
+                }
+                if (stderr && stderr.length > 0) {
+                    reject(new Error(stderr));
+                    return;
+                }
+                resolve(stdout);
+            });
+        });
+    };
+
+    abstract getEntries(): any; //any[]
+    abstract lineParser(line: string): any;
+}
+
+export class DockerPicker extends Picker {
+    getEntries() {
+        var psCommand = ("docker ps --format {{.ID}}@{{.Names}}@{{.Image}}@{{.Labels}}");
+        var _this = this
+        return this.execChildProcess(psCommand, null).then(function (items: string) {
+            return _this.parseItems('\n'+items); //Add bogus header line
+        });
+    }
+
+    lineParser(line: string) {
```

Comment:
Maybe add an error check here in case something goes wrong? This _might_ contain an error message instead of a@b@c@d or it could break in the future or with an old version probably too. Just use `vscode.window.showErrorMessage` to notify the user instead of crashing.